schematic/schematics/redshift_schematic.py

```diff
@@ -0,0 +1,553 @@
+# -*- coding: utf-8 -*-
+# MIT License
+#
+# Copyright (c) 2019 Cody J. Hanson
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""
+Implementation of the schematic.Schematic interface
+with Redshift-specific logic and Schematic.TableColumnType
+implementations.
+
+Based on Redshift documentation:
+- https://docs.aws.amazon.com/redshift/latest/dg/c_Supported_data_types.html
+
+TODO(Cody): Determine how/whether to handle CHAR types (currently defaults to VARCHAR)
+TODO(Cody): Determine how/whether to handle NCHAR and NVARCHAR types
+TODO(Cody): Implement how/whether to handle BPCHAR and TEXT types
+"""
+import schematic
+import re
+from psycopg2 import sql
+
+
+class RedshiftTableColumn(schematic.TableColumn, schematic.NameSqlMixin):
+    """Redshift-specific implementation of TableColumn
+
+    Attributes:
+      distkey: Boolean indicating whether or not this is the distkey
+               for the table.
+      sortkey: Integer indicating which sortkey this is for the table.
+               If this column isn't a sortkey, this is None.
+      encoding: The encoding of this tables values.
+      notnull: Whether or not this column has a NOT NULL constraint.
+    """
+
+    def __init__(self, name, column_type, distkey, sortkey, encoding, notnull):
+        super(RedshiftTableColumn, self).__init__(name, column_type)
```

Comment:
```suggestion
        super().__init__(name, column_type)
```
Unless you're worried about python2